core/crypto/src/signature.rs

```diff
@@ -856,6 +842,27 @@ impl<'de> serde::Deserialize<'de> for Signature {
     }
 }
 
+/// Helper struct which provides Display implementation for bytes slice
+/// encoding them using base58.
+// TODO(mina86): Get rid of it once bs58 has this feature.  There’s currently PR
+// for that.
+struct Bs58<'a>(&'a [u8]);
+
+impl<'a> core::fmt::Display for Bs58<'a> {
+    fn fmt(&self, fmt: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        debug_assert!(self.0.len() <= 65);
+        // The largest buffer we’re ever encoding is 65-byte long.  Base58
+        // increases size of the value by less than 40%.  96-byte buffer is
+        // therefore enough to fit the largest value we’re ever encoding.
+        let mut buf = [0u8; 96];
+        let len = bs58::encode(self.0).into(&mut buf[..]).unwrap();
+        let output = &buf[..len];
```

Comment:
I am a bit uneasy with this length constraint -- `Bs58` has a shape of a general-purpose utility, but it can panic if used wrong. It doesn't seem unlikely that, in two years, someone would use it for something longer than 65 bytes. I'd maybe future-proffed this by using a SmallVec, to support all lengths. I think not doint that is also OK though.

In any case, could you add a unit-test here? (or do we already have tests which exercise Signature's and PK's Displays)?